File: dataset/output-bk/PERPLEXITY_llama-3-sonar-large-32k-chat/Task174_PERPLEXITY_llama-3-sonar-large-32k-chat.java
public class Solution {
    public String shortestPalindrome(String s) {
        String rev_s = new StringBuilder(s).reverse().toString();
        for (int i = 0; i < s.length(); i++) {
            if (s.startsWith(rev_s.substring(i))) {
                return rev_s.substring(0, i) + s;
            }
        }
        return "";
    }
}

File: dataset/output-bk/PERPLEXITY_llama-3-sonar-large-32k-chat/Task174_PERPLEXITY_llama-3-sonar-large-32k-chat.cpp
class Solution {
public:
    string shortestPalindrome(string s) {
        string rev_s = s;
        reverse(rev_s.begin(), rev_s.end());
        for (int i = 0; i < s.length(); i++) {
            if (s.find(rev_s.substr(i)) == 0) {
                return rev_s.substr(0, i) + s;
            }
        }
        return "";
    }
};